### PR TITLE
Adds a method to delete by keypath

### PR DESCRIPTION
**Pull request by yuhanz**
Hi Nick,

I implemented a method to delete by keypath. please merge if you found it desired. thanks.

### CODE REVIEW

**Comment by PikachuEXE** [comment]
Please remove changes about version number
They should only be changed in `master`, not PR

**Comment by yuhanz** [comment]
@PikachuEXE  I removed the  version number changes

**Comment by PikachuEXE** [comment]
Maybe squash the commits into one?
Also the spec is missing.

**Comment by yuhanz** [comment]
squashed all commits into one; specs added for delete_at_path

File: spec/enumerable_extensions_spec.rb [hunk 1]
```diff
@@ -93,4 +93,29 @@
...
+  end
+
+  it 'can delete by path with array index' do
+    source = { items: [{ id: 1 }, {id: 2}, {id: 3}]}
+    expected = {items: [{id: 1}, {id: 3}]}
+    source.delete_at_keypath("items.1")
+    source.must_equal(expected)
+  end
+
+  it 'can delete by each path if the matched is an array of elements' do
```
Comment (PikachuEXE) [inline spec/enumerable_extensions_spec.rb]: I am not sure if that's how `set` works today

**Comment by PikachuEXE** [comment]
I think the spec for "when value is missing" is missing

**Comment by yuhanz** [comment]
added the test case for delete_at_path "when value is missing"

File: spec/enumerable_extensions_spec.rb (hunk 1 above)
Comment (yuhanz) [inline spec/enumerable_extensions_spec.rb]: not quite sure about the set part either

Comment (PikachuEXE) [inline spec/enumerable_extensions_spec.rb]: I don't think this is expected by people who have use `set` before.

Comment (yuhanz) [inline spec/enumerable_extensions_spec.rb]: the bahavior on set is unchanged.

The newly introduced deleting by path on each element is only available on with delete_at_path. 

There wasn't an existing wildcard support. This behavior is implemented as I needed it. and it behaves in a way similar to how mongodb uses key path

Comment (yuhanz) [inline spec/enumerable_extensions_spec.rb]: keypath using array index is identical to the behavior of value_at_keypath

**Comment by tdegrunt** [comment]
Would it be possible to merge this in? I've already merged this in my branch.
https://github.com/tdegrunt/key_path

I'm also willing to take over the maintenance of this gem, perhaps you can add me as an owner for it in ruby gems?

**Comment by nickcharlton** [comment]
Closing as it's old enough to pre-date maintainer pushes by quite a way! It's replaced by #13.

**Pull request closed without merging**